druid/examples/widget_gallery.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use druid::{
+    kurbo::Circle,
+    widget::{
+        prelude::*, Button, Checkbox, Flex, Label, List, Painter, ProgressBar, RadioGroup, Scroll,
+        Slider, Spinner, Stepper, Switch, TextBox,
+    },
+    AppLauncher, Color, Data, Lens, Rect, Widget, WidgetExt, WidgetPod, WindowDesc,
+};
+
+const XI_IMAGE: &[u8] = include_bytes!("assets/xi.image");
+
+#[derive(Clone, Data, Lens)]
+struct AppData {
+    label_data: String,
+    checkbox_data: bool,
+    clicked_count: u64,
+    list_items: Arc<Vec<String>>,
```

Comment:
I guess using `Arc<Vec<String>>` is fine if we don't want to depend on the `im` feature for this example, but I think we should have a comment at least saying that it's better to use `im::Vector` via the `im` feature.

Reply:
I think it would be best to use `im` in all examples where it is applicable, after all it is the recommended way of working with lists in druid right now.

Reply:
Yeah I think changing this to `im::Vector` is the way to go. Initially I was worried about making the example too complex with cfg guards, however they aren't needed. The `im` feature requirement can just be specified via `Cargo.toml` like I did for the other examples in #1050.